hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/StreamWatcher.java

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright 2015-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.alerts.rest;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.ejb.EJB;
+import javax.ejb.Stateless;
+import javax.ws.rs.core.StreamingOutput;
+
+import org.hawkular.alerts.api.json.JsonUtil;
+import org.hawkular.alerts.api.model.event.Alert;
+import org.hawkular.alerts.api.model.event.Event;
+import org.hawkular.alerts.api.model.paging.Order;
+import org.hawkular.alerts.api.model.paging.Page;
+import org.hawkular.alerts.api.model.paging.PageContext;
+import org.hawkular.alerts.api.model.paging.Pager;
+import org.hawkular.alerts.api.services.AlertsCriteria;
+import org.hawkular.alerts.api.services.AlertsService;
+import org.hawkular.alerts.api.services.EventsCriteria;
+import org.jboss.logging.Logger;
+
+/**
+ * Handle StreamingOutput logic for events/alerts Watchers.
+ *
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+@Stateless
+public class StreamWatcher {
+    private static final Logger log = Logger.getLogger(StreamWatcher.class);
+    private static final Pager ctimePager;
+    private static final Pager stimePager;
+    private static final long WATCHER_INTERVAL_DEFAULT = 5000;
+    private static final long CLEAN_INTERVAL = 10000;
+    private static final long LEAP_INTERVAL = 1000;
+
+    @EJB
+    AlertsService alertsService;
+
+    static {
+        List<Order> ordering = new ArrayList<>();
+        ordering.add(Order.by("stime", Order.Direction.ASCENDING));
+        stimePager = new Pager(0, PageContext.UNLIMITED_PAGE_SIZE, ordering);
+        ordering = new ArrayList<>();
+        ordering.add(Order.by("ctime", Order.Direction.ASCENDING));
+        ctimePager = new Pager(0, PageContext.UNLIMITED_PAGE_SIZE, ordering);
+    }
+
+    public StreamingOutput watchAlerts(Set<String> tenantIds, AlertsCriteria criteria, Long watchInterval) {
+        return output -> {
+            Writer writer = new BufferedWriter(new OutputStreamWriter(output));
```

Comment:
The output stream is managed externally and closed by the StreamOutput user.
A close is harmless, anyway, so I will add it.